lightning-background-processor/src/lib.rs

```diff
@@ -43,7 +43,7 @@ pub struct BackgroundProcessor {
 	stop_thread: Arc<AtomicBool>,
 	/// May be used to retrieve and handle the error if `BackgroundProcessor`'s thread
 	/// exits due to an error while persisting.
-	pub thread_handle: JoinHandle<Result<(), std::io::Error>>,
+	pub thread_handle: Option<JoinHandle<Result<(), std::io::Error>>>,
```

Comment:
Maybe we should make this non-pub now? Users can always use `stop` or `stop_and_join_thread`.

Reply:
Yeah, that would be a good idea. This was `pub` to allow waiting on any errors (e.g. from persisting). We'd have to expose a `join` method to allow for that behavior still, which also hides the `Option` implementation detail.

Reply:
Wait, how does that differ from `stop`?

Reply:
Oh, duh.

Reply:
`stop` will stop the thread (if it hasn't already because of an error) and allows you to examine any errors that may have occurred. `join` will wait for any errors to occur. The latter is only useful if your persisters can give an error. I updated the documentation to spell out these cases in terms of `join` and `stop`.